pkg/cache/context.go

```diff
@@ -1621,9 +1621,9 @@ func (ctx *Context) finalizeNodes(existingNodes []*v1.Node) error {
 	}
 
 	// convert the node list into a map
-	nodeMap := make(map[string]*v1.Node)
+	nodeMap := make(map[string]struct{}, len(nodes))
```

Comment:
From a readability perspective I would prefer `nodeMap := make(map[string]bool, len(nodes))` and set `true` in 1626
Then at line 1631 the check simply becomes `if !nodeMap[node.Name] {`

Size (~5% more) and speed wise (not measurable unless the map has millions of entries) it does not make a difference.

Reply:
Updated PR per suggestion.